libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Network/Utilities/CSFInternalDefines.m

```diff
@@ -76,12 +76,8 @@ CSFParameterStyle CSFRequiredParameterStyleForHTTPMethod(NSString *method) {
 Class CSFClassFromEncoding(NSString *encoding) {
     Class result = nil;
     if ([encoding hasPrefix:@"@"]) {
-        if (encoding.length > 2) {
-            NSString *className = [[encoding substringFromIndex:2] stringByReplacingOccurrencesOfString:@"\"" withString:@""];
-            result = NSClassFromString(className);
-        } else {
-            result = [NSObject class];
-        }
+        NSString *className = [[encoding substringFromIndex:2] stringByReplacingOccurrencesOfString:@"\"" withString:@""];
```

Comment:
I'm specifically concerned about the call to `[encoding substringFromIndex:2]` without bounds checking that `encoding` is actually 3 or more characters in length.  I'm find with the business logic change.